File: wasm-module/quelle/prim.c
/* DERSELBE ALGORITHMUS wie kernel/app/prim.fi -- Probedivision.
   Wird nach wasm32-wasi uebersetzt und im Deuter ausgefuehrt. */
#include <stdio.h>
#include <stdlib.h>

static int ist_prim(unsigned long long n) {
    if (n < 2) return 0;
    if (n % 2 == 0) return n == 2;
    for (unsigned long long d = 3; d * d <= n; d += 2)
        if (n % d == 0) return 0;
    return 1;
}

int main(int argc, char **argv) {
    unsigned long long grenze = 200000;
    if (argc > 1) grenze = strtoull(argv[1], 0, 10);
    unsigned long long zahl = 0;
    for (unsigned long long k = 2; k < grenze; k++)
        if (ist_prim(k)) zahl++;
    printf("primzahlen unter %llu: %llu\n", grenze, zahl);
    return 0;
}

File: wasm-module/quelle/prim.rs
// DERSELBE ALGORITHMUS wie kernel/app/prim.fi -- Probedivision.
fn ist_prim(n: u64) -> bool {
    if n < 2 { return false; }
    if n % 2 == 0 { return n == 2; }
    let mut d: u64 = 3;
    while d * d <= n {
        if n % d == 0 { return false; }
        d += 2;
    }
    true
}
fn main() {
    let a: Vec<String> = std::env::args().collect();
    let grenze: u64 = if a.len() > 1 { a[1].parse().unwrap_or(200000) } else { 200000 };
    let mut zahl: u64 = 0;
    let mut k: u64 = 2;
    while k < grenze {
        if ist_prim(k) { zahl += 1; }
        k += 1;
    }
    println!("primzahlen unter {}: {}", grenze, zahl);
}
